src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -197,50 +193,12 @@ private void visitObject(MaskingState maskingState, @CheckForNull KeyMaskingConf
      * @param keyMaskingConfig the {@link KeyMaskingConfig} for the corresponding JSON key
      */
     private void maskString(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
-        maskingState.incrementCurrentIndex(); // step over the string value opening quote
-        int targetValueLength = 0;
-        int noOfEscapeCharacters = 0;
-        int additionalBytesForEncoding = 0;
-        boolean isEscapeCharacter = false;
-        boolean previousCharacterCountedAsEscapeCharacter = false;
-        while (!AsciiCharacter.isDoubleQuote(maskingState.byteAtCurrentIndex()) || (
-                AsciiCharacter.isDoubleQuote(maskingState.byteAtCurrentIndex())
-                        && isEscapeCharacter)) {
-            if (Utf8Util.getCodePointByteLength(maskingState.byteAtCurrentIndex()) > 1) {
-                /*
-                 * We only support UTF-8, so whenever code points are encoded using multiple bytes this should
-                 * be represented by a single asterisk and the additional bytes used for encoding need to be
-                 * removed.
-                 */
-                additionalBytesForEncoding += Utf8Util.getCodePointByteLength(maskingState.byteAtCurrentIndex()) - 1;
-            }
-            isEscapeCharacter =
-                    AsciiCharacter.isEscapeCharacter(maskingState.byteAtCurrentIndex())
-                            && !previousCharacterCountedAsEscapeCharacter;
-            if (isEscapeCharacter) {
-                /*
-                 * Non-escaped backslashes are escape characters and are not actually part of the string but
-                 * only used for character encoding, so must not be included in the mask.
-                 */
-                noOfEscapeCharacters++;
-                previousCharacterCountedAsEscapeCharacter = true;
-            } else {
-                if (previousCharacterCountedAsEscapeCharacter
-                        && AsciiCharacter.isLowercaseU(maskingState.byteAtCurrentIndex())) {
-                    /*
-                     * The next 4 characters are hexadecimal digits which form a single character and are only
-                     * there for encoding, so must not be included in the mask.
-                     */
-                    additionalBytesForEncoding += 4;
-                }
-                previousCharacterCountedAsEscapeCharacter = false;
-            }
-            targetValueLength++;
-            maskingState.incrementCurrentIndex();
-        }
+        int valueStartIndex = maskingState.currentIndex();
+        skipStringValue(maskingState);
+        int targetValueLength = maskingState.currentIndex() - valueStartIndex;
```

Comment:
This definitely requires one or two lines of documentation 😄 
It is not super obvious to see why you jump over the String to mask it

Reply:
Actually, now that I think of it. I think we should rename all `skip[Type]Value` methods with `stepOver[Type]Value` that would make it clear for both cases right away what is happening. WDYT?

Reply:
I think stepOver sounds good